File: lista-encadeada.py
class No:
    def __init__(self, value):
        self.value = value
        self.next = None


class Linked_list:
    def __init__(self):
        self.first = No(None)

    def insert(self, value):
        if self.first.value is None:
            self.first = No(value)
        else:
            aux = self.first
            while aux.next is not None:
                aux = aux.next
            aux.next = No(value)

    def remove(self, value):
        if self.first.value is None:
            return None
        else:
            aux = self.first
            aux_before = aux
            while aux.next is not None:
                if aux.value == value:
                    aux_before.next = aux.next
                    break
                else:
                    aux_before = aux
                    aux = aux.next

File: listaEncadeada.js
/*
Copyright: GRIJP, Jean.
(04/2020)

*/
class No {
  constructor(value) {
    this.value = value;
    this.next = null;
  }
}

class LinkedList {
  constructor() {
    this.first = new No(null);
  }

  insert(value) {
    if (this.first.value === null) {
      this.first = new No(value);
    } else {
      let aux = this.first;
      while (aux.next !== null) {
        aux = aux.next;
      }
      aux.next = new No(value);
    }
  }

  remove(value) {
    if (this.first.value === null) {
      return null;
    } else {
      let aux = this.first;
      let auxBefore = aux;
      while (aux.next !== null) {
        if (aux.value === value) {
          auxBefore.next = aux.next;
        } else {
          auxBefore = aux;
          aux = aux.next;
        }
      }
    }
  }
}
